### PR TITLE
Fixed declare

### LINKED ISSUE #786: Make declare test work
https://github.com/foundry-rs/starknet-foundry/blob/4a5b36e4d755eec31ba886ef61c485f3a25726dd/crates/cheatnet/tests/cheatcodes/declare.rs#L40

We shouldn't execute it as a transaction from now

### PR DESCRIPTION
**Pull request by Radinyn**
Closes #786

## Introduced changes

- Declare now works the same way as before, by setting the compiled class hash  value to 0.

### CODE REVIEW

File: crates/cheatnet/src/cheatcodes/declare.rs [hunk 1]
```diff
@@ -39,45 +40,31 @@ impl BlockifierState<'_> {
...
-        //         Err(TransactionExecutionError::DeclareTransactionError { class_hash })
-        //     }
-        // }
-
-        Ok(class_hash)
+                // NOTE: Compiled class hash is being set to 0 here
+                blockifier_state
+                    .set_compiled_class_hash(class_hash, Default::default())
+                    .unwrap_or_else(|err| panic!("Failed to set compiled class hash: {err:?}"));
+                Ok(class_hash)
```
Comment (piotmag769) [inline crates/cheatnet/src/cheatcodes/declare.rs:55]: Cannot we compute real compiled class hash from CASM using blockifier's api?

Comment (war-in) [reply crates/cheatnet/src/cheatcodes/declare.rs:55]: Do we need this for anything?

**Review by war-in — APPROVED** [review]

File: crates/cheatnet/src/cheatcodes/declare.rs [hunk 2]
```diff
@@ -39,45 +40,31 @@ impl BlockifierState<'_> {
...
-        //     Err(error) => Err(error).map_err(TransactionExecutionError::from),
-        //     Ok(_) => {
-        //         // Class is already declared, cannot redeclare
-        //         // (i.e., make sure the leaf is uninitialized).
-        //         Err(TransactionExecutionError::DeclareTransactionError { class_hash })
-        //     }
-        // }
-
-        Ok(class_hash)
+                // NOTE: Compiled class hash is being set to 0 here
```
Comment (Arcticae) [inline crates/cheatnet/src/cheatcodes/declare.rs:51]: Please comment "why" - instead of "what"

File: crates/cheatnet/src/cheatcodes/declare.rs [hunk 3]
```diff
@@ -39,45 +40,31 @@ impl BlockifierState<'_> {
...
+            Err(error) => Err(CheatcodeError::Unrecoverable(EnhancedHintError::State(
+                error,
+            ))),
+            Ok(_) => {
+                // Class is already declared, cannot redeclare
+                // (i.e., make sure the leaf is uninitialized).
+                Err(CheatcodeError::Unrecoverable(EnhancedHintError::Anyhow(
+                    anyhow!("Class hash {} is already declared", class_hash),
+                )))
+            }
```
Comment (Arcticae) [inline crates/cheatnet/src/cheatcodes/declare.rs:66]: Are those errors tested? I think intuitively, errors should be different now than the ones from the txn flow, so perhaps we should test more thoroughly the handling

**Review by Arcticae — CHANGES_REQUESTED** [review]

File: crates/cheatnet/src/cheatcodes/declare.rs (hunk 3 above)
Comment (Radinyn) [reply crates/cheatnet/src/cheatcodes/declare.rs:66]: The "already declared" error is tested

File: crates/cheatnet/src/cheatcodes/declare.rs (hunk 1 above)
Comment (piotmag769) [reply crates/cheatnet/src/cheatcodes/declare.rs:55]: Since we communicate that we are consistent with starknet - it would be nice

**Review by Arcticae — APPROVED** [review]

**Pull request merged**